elasticgraph-indexer/spec/acceptance/derived_indexing_types_spec.rb

```diff
@@ -32,7 +34,7 @@ module ElasticGraph
         "nested_fields" => {
           "max_widget_cost" => w1.fetch("record").fetch("cost").fetch("amount_cents")
         },
-        "oldest_widget_created_at" => w1.fetch("record").fetch("created_at")
+        "oldest_widget_created_at" => normalize_datetime(w1.fetch("record").fetch("created_at"))
```

Comment:
Instead of simply normalizing the datetime in the expectations of these acceptance tests, can we pick specific `created_at` values that reproduce the bug your PR is fixing? For example, can we index a widget with `created_at = 2025-01-15T10:30:00.531Z`, then index a widget with `created_at = 2025-01-15T10:30:00.53Z`, then assert that the `oldest_widget_created_at` value is `2025-01-15T10:30:00.53Z` (or `2025-01-15T10:30:00.530Z`) rather than `2025-01-15T10:30:00.531Z`.

My hope is to have the acceptance test here demonstrate _why_ we the precision normalization is needed. Simply asserting that the returned `DateTime` values have 3 digits doesn't do that.